src/sidebar/components/MarkdownEditor.tsx

```diff
@@ -203,25 +203,26 @@ function TextArea({
 
     const textarea = textareaRef.current!;
     const listenerCollection = new ListenerCollection();
+    const checkShouldOpenPopover = () => {
+      const term = termBeforePosition(textarea.value, textarea.selectionStart);
+      setPopoverOpen(term.startsWith('@'));
+    };
 
     // We listen for `keyup` to make sure the text in the textarea reflects the
     // just-pressed key when we evaluate it
     listenerCollection.add(textarea, 'keyup', e => {
       // `Esc` key is used to close the popover. Do nothing and let users close
       // it that way, even if the caret is in a mention
-      if (e.key === 'Escape') {
-        return;
+      if (e.key !== 'Escape') {
+        checkShouldOpenPopover();
       }
-      setPopoverOpen(
-        termBeforePosition(textarea.value, textarea.selectionStart).startsWith(
-          '@',
-        ),
-      );
     });
 
-    return () => {
-      listenerCollection.removeAll();
-    };
+    // When clicking the textarea it's possible the caret is moved "into" a
+    // mention, so we check if the popover should be opened
+    listenerCollection.add(textarea, 'click', checkShouldOpenPopover);
```

Comment:
Does this work if you move the caret with long-press on mobile?

Reply:
Good point. Let me check.